sympy/solvers/bivariate.py

```diff
@@ -196,6 +198,40 @@ def _solve_lambert(f, symbol, gens):
       a = -1, d = a*log(p), f = -log(d) - g*log(p)
     """
 
+    def _solve_even_degree_expr(expr, symbol):
+        """
+        Helper function to solve an expr containing dummy variable t at even
+        degree places to replace them with +symbol and -symbol.
+
+        For example:
+        Equation 2*log(t) + x/2 - log(a) = 0 will return solutions
```

Comment:
Here are some ideas:

```python
def _solve_even_degree_expr(expr, symbol):
    """helper to replace ``t`` with ``+/- symbol`` in ``expr`` and
     return a list of all solutions.

    For example, if ``expr = 2*log(t) + x/2` then solutions for ``2*log(x) + x/2 = 0`` and
    ``2*log(-x) + x/2 = 0`` would be returned.

    For an expression like ``eq = x**2*g(x) = 1``, if we take the log of both sides we obtain
    ``log(x**2) + log(g(x)) = 0``. If x is positive then this simplifies to ``2*log(x) + log(g(x))``;
    the Lambert-solving routines will return solutions for this, but we must also consider
    the solutions for  ``2*log(-x) + log(g(x))`` since those must also be a solution of ``eq``
    which has the same value whether the ``x`` in ``x**2`` is negated. If `g(x)` is not even
    then we don't want to replace the ``x`` there with ``-x``. So the role of the ``t`` in the
    expression received by this function is to mark where ``+/-x`` should be inserted
    before obtaining the Lambert solutions.
```

Reply:
> This sentence doesn't really makes sense is @smichr 's comment addressed?

Which comment is not addressed?
> Here are some ideas:

Thanks @smichr :raised_hands:

Reply:
Since the lambert solver will already handle `log(abs(x))` properly, instead of using `t` and having a special solver for that, how about just replacing even powers as `lambda x: abs(x.base)**x.exp` instead of `lambda x: t**x.exp` and continue with the solving process per normal?

Reply:
Yeah this thought strike on my mind long before. But there was some problem like this:
```python
>>> lhs = log(x**2 + 1) + x
>>> (lhs).replace(lambda i:i.is_Pow and i.base == x and i.exp%2==0, lambda i: abs(x)**i.exp)
x + log(x**2 + 1)
# expected -> x + log(abs(x)**2 + 1)

Reply:
And one of the problem that I see of `_solve_lambert` is that it doesn't return solution unless the symbol is positive in it(From where I think the whole problem originates).
```python
>>> from sympy.solvers.bivariate import _filtered_gens, _solve_lambert, _lambert
>>> x = symbols('x')
>>> eq = x*exp(x/2) + a
>>> _solve_lambert(eq, x, _filtered_gens(Poly(eq), x))
NotImplementedError
>>> x = symbols('x', positive=True)
[2*LambertW(-a/2)]
```
By python debugging on solve it was come to know these points about `_solve_lambert`:
1. Solve first attempts to feed the equation with sybol with predefined assumptions in `_solve_lambert` and it raises NotImplementedError.
2. Then a second attempt is made by `solve` in which the symbol in the equation are made to has assumption positive. By which it returns solution.

According to me when this function was created they ignored the fact(or it is designed the way it is) that in Sympy does not convert `log(x**2) + log(g(x)) = 0` to `2*log(x) + log(g(x)) = 0` unless the symbol is assumed to be positive and thats why `log(x**2) + log(g(x)) = 0` when goes in `_lambert` gives `NotImplementedError` as it was not of the form `a*log(b*X + c) + d*X - f = 0`.

Reply:
And the general behaviour that should be while handling log in equation should be:
1. x**even*g(x) - 1 --> even*log(abs(x)) + log(g(x))
2. x**odd*g(x) - 1 --> odd*log(x) + log(g(x))

But there is no need to indulge in this complexity. `x**power*g(x) - 1` should return as ` log(x**power) + log(g(x))` as it is currently. Because it is the disability of `_solve_lambert` to not able to identify the type.